chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/SpringBasedHttpClient.java

```diff
@@ -0,0 +1,161 @@
+package com.griddynamics.jagger.invoker.http.v2;
+
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.web.client.ResponseErrorHandler;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriTemplateHandler;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.griddynamics.jagger.invoker.http.v2.JHttpEndpoint.Protocol.HTTPS;
+import static com.griddynamics.jagger.invoker.http.v2.SpringBasedHttpClient.JSpringBasedHttpClientParameters.DEFAULT_URI_VARIABLES;
+import static com.griddynamics.jagger.invoker.http.v2.SpringBasedHttpClient.JSpringBasedHttpClientParameters.ERROR_HANDLER;
+import static com.griddynamics.jagger.invoker.http.v2.SpringBasedHttpClient.JSpringBasedHttpClientParameters.INTERCEPTORS;
+import static com.griddynamics.jagger.invoker.http.v2.SpringBasedHttpClient.JSpringBasedHttpClientParameters.MESSAGE_CONVERTERS;
+import static com.griddynamics.jagger.invoker.http.v2.SpringBasedHttpClient.JSpringBasedHttpClientParameters.REQUEST_FACTORY;
+import static com.griddynamics.jagger.invoker.http.v2.SpringBasedHttpClient.JSpringBasedHttpClientParameters.URI_TEMPLATE_HANDLER;
+import static java.lang.String.format;
+
+/**
+ * Implementation of {@link JHttpClient}. <p>
+ * This implementation is based on the Spring {@link RestTemplate}.
+ *
+ * @author Anton Antonenko
+ * @see JHttpClient
+ * @since 1.3
+ */
+@SuppressWarnings({"unused", "unchecked"})
+public class SpringBasedHttpClient implements JHttpClient {
+
+    /**
+     * values: {@link JSpringBasedHttpClientParameters#DEFAULT_URI_VARIABLES}, {@link JSpringBasedHttpClientParameters#ERROR_HANDLER},
+     * {@link JSpringBasedHttpClientParameters#MESSAGE_CONVERTERS}, {@link JSpringBasedHttpClientParameters#URI_TEMPLATE_HANDLER},
+     * {@link JSpringBasedHttpClientParameters#INTERCEPTORS}, {@link JSpringBasedHttpClientParameters#REQUEST_FACTORY}
+     */
+    public enum JSpringBasedHttpClientParameters {
```

Comment:
What is client params? can you explain here? As I understood it is some kind of configuration params.